src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertyAccessorSymbol.cs

```diff
@@ -451,6 +456,11 @@ private void CheckModifiers(Location location, bool hasBody, bool isAutoProperty
             {
                 diagnostics.Add(AccessCheck.GetProtectedMemberInSealedTypeError(ContainingType), location, this);
             }
+            else if (IsStatic && IsReadOnly && !_property.HasReadOnlyModifier)
```

Comment:
>&& !_property.HasReadOnlyModifier [](start = 44, length = 33)

Is this check needed?

Reply:
In this scenario: `static readonly int P { get; set; }` this check allows us to give a diagnostic only on the property and not on the accessors, which inherit their modifiers from the property. (is there a better term to use than "inherit"?)

Reply:
Is there a better pattern to use to prevent redundant diagnostics?